fastMatcher.go

```diff
@@ -265,28 +266,39 @@ func (m *FastMatcher) matchOp(op *OpNode, litVal *FastVal) error {
 	}
 
 	var opRes bool
+	var validOp bool
+	var compareOut int
 	switch op.Op {
 	case OpTypeEquals:
-		opRes = lhsVal.Equals(rhsVal)
+		opRes, validOp = lhsVal.Equals(rhsVal)
 	case OpTypeLessThan:
-		opRes = lhsVal.Compare(rhsVal) < 0
+		compareOut, validOp = lhsVal.Compare(rhsVal)
+		opRes = compareOut < 0
 	case OpTypeLessEquals:
-		opRes = lhsVal.Compare(rhsVal) <= 0
+		compareOut, validOp = lhsVal.Compare(rhsVal)
+		opRes = compareOut <= 0
 	case OpTypeGreaterThan:
-		opRes = lhsVal.Compare(rhsVal) > 0
+		compareOut, validOp = lhsVal.Compare(rhsVal)
+		opRes = compareOut > 0
 	case OpTypeGreaterEquals:
-		opRes = lhsVal.Compare(rhsVal) >= 0
+		compareOut, validOp = lhsVal.Compare(rhsVal)
+		opRes = compareOut >= 0
 	case OpTypeMatches:
-		opRes = lhsVal.Matches(rhsVal)
+		opRes, validOp = lhsVal.Matches(rhsVal)
 	case OpTypeExists:
 		opRes = true
+		validOp = true
 	default:
 		panic("invalid op type")
 	}
 
 	// Mark the result of this operation
 	m.buckets.MarkNode(bucketIdx, opRes)
```

Comment:
Will not be any functional issue by storing the opRes irrespective of validOp status?

Reply:
No issues.